packages/build-tools/src/generic.ts

```diff
@@ -0,0 +1,49 @@
+import path from 'path';
+
+import { BuildPhase, Generic } from '@expo/eas-build-job';
+import { BuildConfigParser, BuildStepGlobalContext, errors } from '@expo/steps';
+import nullthrows from 'nullthrows';
+
+import { BuildContext } from './context';
+import { prepareProjectSourcesAsync } from './common/projectSources';
+import { getEasFunctions } from './steps/easFunctions';
+import { CustomBuildContext } from './customBuildContext';
+import { getEasFunctionGroups } from './steps/easFunctionGroups';
+
+export async function runGenericJobAsync(ctx: BuildContext<Generic.Job>): Promise<void> {
+  const customBuildCtx = new CustomBuildContext(ctx);
+
+  await ctx.runBuildPhase(BuildPhase.PREPARE_PROJECT, async () => {
+    await prepareProjectSourcesAsync(ctx, customBuildCtx.projectSourceDirectory);
+  });
```

Comment:
To make it more interesting/interactive for a user it would be good if it printed some logs because this phase will be displayed on the website anyway, but as an empty phase. 

If we add logging to this function we should be aware that we probably would need to wrap it in a build phase for normal builds as well or only print logs for generic jobs.

Reply:
It can also be confusing for the users as normally the archive is downloaded during the `checkout` function run. Our checkout is weird and that could have been a reason why this was kind of hidden 🤔

Reply:
Hmm, sure. Removed wrapper.